src/pkg/task/dao/execution.go

```diff
@@ -132,3 +137,63 @@ func (e *executionDAO) Delete(ctx context.Context, id int64) error {
 	}
 	return nil
 }
+
+func (e *executionDAO) RefreshStatus(ctx context.Context, id int64) (err error) {
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return err
+	}
+	// set a score for each status:
+	// 		pending, running, scheduled - 4
+	// 		error - 3
+	//		stopped - 2
+	//		success - 1
+	// and set the status of record with highest score as the status of execution
+	sql := `with status_score as (
```

Comment:
SQL is a bit complicated and hard to maintain, can we split the logic as multiple steps/logics?

Reply:
Use a single one sql to avoid the concurrent issue, will add more comment to explain the sql